### PR TITLE
Fixes #18434: Implement ips package repositories technique

### PR DESCRIPTION
**Pull request by peckpeck**
https://issues.rudder.io/issues/18434

### CODE REVIEW

File: techniques/applications/ipsPackageRepositories/1.0/ipsPackageRepositories.st [hunk 1]
```diff
@@ -0,0 +1,48 @@
+bundle agent ips_repository_&RudderUniqueID& {
```
Comment (amousset) [inline techniques/applications/ipsPackageRepositories/1.0/ipsPackageRepositories.st:1]: missing licence

File: techniques/applications/ipsPackageRepositories/1.0/ipsPackageRepositories.st [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  methods:
+    "ips_repository" usebundle => ips_repository_real_&RudderUniqueID&("&IPS_PUBLISHER_NAME&", "&IPS_PUBLISHER_URL&", "&IPS_PUBLISHER_ENABLED&");
```
Comment (amousset) [inline techniques/applications/ipsPackageRepositories/1.0/ipsPackageRepositories.st:3]: I like this

File: techniques/applications/ipsPackageRepositories/1.0/ipsPackageRepositories.st [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    "args"    slist => { ${name}, ${url}, ${enabled} };
+    "prefix" string => "Publisher check_${report_data.directive_id}";
+
+  methods:
+    "${prefix}_0" usebundle => _method_reporting_context("Publisher check", "publisher_exist_${name}");
+    "${prefix}_0" usebundle => condition_from_command("publisher_exist_${name}", "pkg publisher ${name} >/dev/null 2>/dev/null", "0", "1");
+
+    "${prefix}_1" usebundle => _method_reporting_context("Publisher create", "pkg set-publisher -g \"${url}\" \"${name}\"");
+    "${prefix}_1" usebundle => command_execution("pkg set-publisher -g \"${url}\" \"${name}\""),
+                                                            if => concat("condition_from_command_publisher_exist_",canonify("${name}"),"_false");
```
Comment (amousset) [inline techniques/applications/ipsPackageRepositories/1.0/ipsPackageRepositories.st:20]: if is not aligned

Comment (amousset) [reply techniques/applications/ipsPackageRepositories/1.0/ipsPackageRepositories.st:20]: same for all other conditions

File: techniques/applications/ipsPackageRepositories/1.0/metadata.xml [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        <LONGDESCRIPTION>Name of the publisher repository</LONGDESCRIPTION>
+      </INPUT>
+      <INPUT>
+        <NAME>IPS_PUBLISHER_URL</NAME>
+        <DESCRIPTION>URL</DESCRIPTION>
+        <LONGDESCRIPTION>URL of the publisher repository</LONGDESCRIPTION>
+      </INPUT>
+      <SELECT1>
+        <NAME>IPS_PUBLISHER_ENABLED</NAME>
+        <DESCRIPTION>Enabled</DESCRIPTION>
```
Comment (amousset) [inline techniques/applications/ipsPackageRepositories/1.0/metadata.xml:50]: Maybe "State" more than "Enabled" which a value

Comment (amousset) [reply techniques/applications/ipsPackageRepositories/1.0/metadata.xml:50]: And option should be "Enabled" and "Disabled" to match generic methods semantics

**Comment by peckpeck** [comment]
PR updated with a new commit

**Comment by peckpeck** [comment]
PR updated with a new commit

**Review by amousset — APPROVED** [review]

**Comment by peckpeck** [comment]
OK, squash merging this PR

**Pull request merged**